tldraw/packages/react/src/components/Canvas/Canvas.tsx

```diff
@@ -111,119 +112,200 @@ export const Canvas = observer(function Renderer<S extends TLReactShape>({
+                    >
+                      <components.SelectionForeground
+                        zoom={zoom}
+                        shapes={selectedShapes}
+                        bounds={selectionBounds}
+                        showResizeHandles={showResizeHandles}
+                        showRotateHandles={showRotateHandles}
+                      />
+                    </Container>
+                  )}
+                  {showHandles && onlySelectedShapeWithHandles && components.Handle && (
+                    <Container
+                      data-type="onlySelectedShapeWithHandles"
+                      bounds={selectionBounds}
+                      zIndex={10003}
+                    >
+                      <SVGContainer>
+                        {Object.entries(onlySelectedShapeWithHandles.props.handles!).map(
+                          ([id, handle]) =>
+                            React.createElement(components.Handle!, {
+                              key: `${handle.id}_handle_${handle.id}`,
+                              shape: onlySelectedShapeWithHandles,
+                              handle,
+                              id,
+                            })
+                        )}
+                      </SVGContainer>
+                    </Container>
+                  )}
+                  {selectedShapes && components.SelectionDetail && (
+                    <SelectionDetailContainer
+                      key={'detail' + selectedShapes.map(shape => shape.id).join('')}
+                      shapes={selectedShapes}
+                      bounds={selectionBounds}
+                      detail={showSelectionRotation ? 'rotation' : 'size'}
+                      hidden={!showSelectionDetail}
+                      rotation={selectionRotation}
+                    />
+                  )}
+                  {selectedShapes && components.ContextBar && (
+                    <ContextBarContainer
+                      key={'context' + selectedShapes.map(shape => shape.id).join('')}
+                      shapes={selectedShapes}
+                      hidden={!showContextBar}
+                      bounds={
+                        selectedShapes.length === 1 ? selectedShapes[0].bounds : selectionBounds
+                      }
+                      rotation={selectedShapes.length === 1 ? selectedShapes[0].props.rotation : 0}
+                    />
+                  )}
+                </>
               )}
+            </HTMLLayer>
+            {selectionDirectionHint && selectionBounds && selectedShapes && (
+              <DirectionIndicator
+                direction={selectionDirectionHint}
+                bounds={selectionBounds}
+                shapes={selectedShapes}
+              />
+            )}
+
+            <div id="tl-dev-tools-canvas-anchor" />
+          </div>
+          {children}
+        </ContextMenu.Trigger>
+        <ContextMenu.Content className="tl-context-menu">
```

Comment:
Can you move out the context menu related components to a another file? Just like context bar component.